pkgs/servers/web-apps/lemmy/ui.nix

```
@@ -0,0 +1,82 @@
+{ lib
+, mkYarnPackage
+, libsass
+, nodejs
+, python3
+, pkg-config
+, writeShellScriptBin
+, jq
+, runCommand
+, fetchFromGitHub
+}:
+
+let
+  pkgConfig = {
+    node-sass = {
+      nativeBuildInputs = [ ];
+      buildInputs = [ libsass pkg-config python3 ];
+      postInstall = ''
+        LIBSASS_EXT=auto yarn --offline run build
+        rm build/config.gypi
+      '';
+
+      dontFixup = false;
+
+    };
+  };
+
+  pname = "lemmy-ui";
+  version = "0.12.2";
+  src = fetchFromGitHub {
+    owner = "LemmyNet";
+    repo = pname;
+    rev = "${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-iFLJqUnz4m9/JTSaJSUugzY5KkiKtH0sMYY4ALm2Ebk=";
+  };
+
+  patchedPackageJSON = runCommand "package.json" { } ''
+    ${jq}/bin/jq '.version = "${version}"' ${src}/package.json > $out
+  '';
+
+  unwrapped = mkYarnPackage rec {
+    name = "${pname}-${version}";
+
+    inherit src version pkgConfig;
+
+    extraBuildInputs = [ libsass ];
+
+    yarnNix = ./yarn.nix;
+
+    packageJSON = patchedPackageJSON;
+
+    yarnPreBuild = ''
+      export npm_config_nodedir=${nodejs}
+    '';
+
+    buildPhase = ''
+      # Yarn writes cache directories etc to $HOME.
+      export HOME=$PWD/yarn_home
+
+      ln -sf $PWD/node_modules $PWD/deps/lemmy-ui/
+
+      yarn --offline build:prod
+    '';
+
+    distPhase = "true";
+
+    meta = with lib; {
+      description = "🐀 Building a federated alternative to reddit in rust";
```

probably want to avoid unicode
```suggestion
      description = "Building a federated alternative to reddit in rust";
```